src/Nethermind/Nethermind.Blockchain/Comparers/GasPriceTxComparer.cs

```diff
@@ -0,0 +1,41 @@
+﻿//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System.Collections.Generic;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+
+namespace Nethermind.Blockchain.Comparers
+{
+    public class GasPriceTxComparer : IComparer<Transaction>
+    {
+        private readonly IBlockTree _blockTree;
+        private readonly IGasPriceTxComparerByBaseFee _gasPriceTxComparerByBaseFee;
+
+        public GasPriceTxComparer(IBlockTree blockTree, IGasPriceTxComparerByBaseFee gasPriceTxComparerByBaseFee)
+        {
```

Comment:
a bit confusing, does this wrapper add any logic?

Reply:
I agree with you that it was confusing. I changed it a little and hope it is clear now.
We have two different sources of base fee:
1. From block tree head block (GasPriceTxComparer)
2. From preparing block context (GasPriceTxComparerForProducer)

But these two situations have shared logic (GasPriceTxComparerHelper)